tests/mypy_self_check.py

```diff
@@ -2,17 +2,29 @@
 """Script to run mypy against its own code base."""
 
 import os
+import re
 import subprocess
 import sys
 import tempfile
 from pathlib import Path
 
-# Use the current mypy version until a version that supports modular
-# typeshed is released on PyPI.
-MYPY_VERSION = "git+git://github.com/python/mypy"
+REQUIREMENTS_FILE = "requirements-tests-py3.txt"
+MYPY_REQUIREMENTS_REGEXPS = [r"^mypy[ =>]", r"^git\+.*/mypy.git\W"]
+
+
+def determine_mypy_version() -> str:
+    with open(REQUIREMENTS_FILE) as f:
+        for line in f:
+            for regexp in MYPY_REQUIREMENTS_REGEXPS:
+                m = re.match(regexp, line)
+                if m:
+                    return line.strip()
+    raise RuntimeError(f"no mypy version found in {REQUIREMENTS_FILE}")
```

Comment:
Can we just install the whole `requirements-tests-py3.txt`? It still wouldn't be the slowest thing in our CI.

Reply:
Fine by me and it would make this less error prone.

Reply:
@Akuli I just wanted to make that change, but noticed that after the installation of mypy, mypy's test-requirements.txt file is installed, which may contain conflicting package versions. I think it's safer to just install mypy.

Reply:
The code still seems unnecessarily complicated to me. Currently `.github/workflows/tests.yml` contains commands like `pip install $(grep black requirements-tests-py3.txt)`. Maybe we could similarly switch to a simple `if "mypy" in line`?

That said, this is just a small nit, and I'm fine with it either way.

Reply:
I was considering using something simple, but a simple check would match comments and things like "flake8-mypy". (The latter is less of a concern at the moment, though.)

Reply:
Actually, `"mypy" in line` breaks because the file now has a comment mentioning mypy.